docs/lensvm/lens-vm.md

```diff
@@ -0,0 +1,161 @@
+  - Please pay attention to any prerequisites, for example if on Ubuntu you may need to install the `build-essential` package
+- If connection errors are experienced when retrieving rust package dependencies from crates.io, you might need to tweak your `.gitconfig` as per this [comment](https://github.com/rust-lang/cargo/issues/3381#issuecomment-1193730972).
+- `npm`, typically installed via [nvm](https://github.com/nvm-sh/nvm#install--update-script)
+- [Go](https://golang.google.cn/doc/install)
+
+## Basic Lens Example
+
+The easiest way to get started writing a Lens is by using Rust, thanks to the [`lens_sdk`](https://docs.rs/lens_sdk/0.8.0/lens_sdk) crate, which provides helpful macros and utilities for Lens development.
+
+A minimal example is shown in the [`define!` macro documentation](https://docs.rs/lens_sdk/0.8.0/lens_sdk/macro.define.html#examples). This example demonstrates a simple forward transformation that iterates through input documents and increments the `age` field by 1.
+
+## Writing a Lens (SDK)
+
+Writing a simple Lens using the Rust SDK is straightforward, and the process is well documented in the [`lens_sdk` crate documentation](https://docs.rs/lens_sdk/0.8.0/lens_sdk).
+
+As shown in the [Basic Lens Example](#basic-lens-example), the first example demonstrates a minimal forward transformation. The second example in the [`define!` macro docs](https://docs.rs/lens_sdk/0.8.0/lens_sdk/macro.define.html#examples) builds upon this by introducing parameters and an inverse function, showcasing how Lenses can support bi-directional transformations.
+
+For more real-world and test-driven examples, explore the following repositories:
+
+- [Lens test modules](https://github.com/lens-vm/lens/blob/main/tests/modules)
+- [DefraDB lens tests](https://github.com/sourcenetwork/defradb/tree/develop/tests/lenses)
+
+These examples provide additional context and cover a range of use cases, including schema-aware transformations and reversible pipelines.
+
+## Writing a Lens (without SDK)
+
+Writing a Lens without using the Rust SDK is significantly more involved and intended for advanced users who want to build in languages other than Rust, or prefer full control over serialization and transformation logic.
+
+Currently, the only working example we have is written in [AssemblyScript](https://www.assemblyscript.org/introduction.html). You can find it here:
+
+- [AssemblyScript Lens example](https://github.com/lens-vm/lens/blob/main/tests/modules/as_wasm32_simple/assembly/index.ts)
+
+This example demonstrates a minimal Lens implementation without the support of an SDK. However, it requires:
+
+- A deep understanding of the Lens transfer protocol
+- Manual implementation of (de)serialization logic
+- Proficiency in AssemblyScript or your language of choice
+
+**Note:** The AssemblyScript example was created as a proof of concept and may not reflect best practices. It was also the author’s first and only project in that language.
```

Comment:
suggestion: Including `It was also the author’s first and only project in that language.` unnecessarily advertises that we don't know what we are doing - the first sentence alone is sufficient IMO.